src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -301,6 +306,34 @@ def is_saveable_module(name, value):
 
             save_method(os.path.join(save_directory, pipeline_component_name), **save_kwargs)
 
+            if dduf_format:
+                import shutil
+                import tarfile
+
+                dduf_file_path = os.path.join(save_directory, dduf_filename)
+
+                if os.path.isdir(dduf_file_path):
+                    logger.warning(
+                        f"Removing the existing folder {dduf_file_path} so that we can save the DDUF archive."
+                    )
+                    shutil.rmtree(dduf_file_path)
+                if (
+                    os.path.exists(dduf_file_path)
+                    and os.path.isfile(dduf_file_path)
+                    and tarfile.is_tarfile(dduf_file_path)
+                ):
+                    # Open in append mode if the file exists
+                    mode = "a"
+                else:
+                    # Open in write mode to create it if it doesn't exist
+                    mode = "w:"
+                with tarfile.open(dduf_file_path, mode) as tar:
+                    dir_to_archive = os.path.join(save_directory, pipeline_component_name)
+                    if os.path.isdir(dir_to_archive):
```

Comment:
We also rely on `model_index.json` to help with loading and a few verifications. We should package that in the tar archive, IMO. Any reason not to?

Reply:
I agree with Sayak, `model_index.json` can serve as a first approach to containing pipeline metadata (Even if it's duplicated in the repo as a separate file anyway).

Reply:
Sounds good, I'll package it !

Reply:
Done in the latest commit